### ISSUE TITLE
Variable to check for web view vs. email

### ISSUE THREAD

**Issue by luke-nehemedia** [putyourlightson/craft-campaign#246]
A client of mine often includes videos in their newsletters. Since this is not really possible in an email, I only include a cover-image of the video with a link to the web-view of the newsletter in place of the video. When the user clicks on the link it opens the web-view and the video is loaded in place of the cover-image.

To implement this, I have to check whether the current render is a web-view or the email-view. It would be very handy to have a variable in place like `isWebRequest` to check for this.

Right now I use the following code:

```twig
{% if craft.app.controller is instance of('craft\\controllers\\TemplatesController') %}
      <mjml...>display video</mjml>
{% elseif (craft.app.controller is instance of ('craft\\controllers\\QueueController')) or (craft.app.controller is instance of ('putyourlightson\\campaign\\controllers\\SendoutsController')) %}
    <mj-image href="{{browserVersionUrl}}" target="_blank" /> Link to web-version</mjml-image>
{% endif %}
```

This works for the sendout-preview and the sent mail, but not for test-mails. Therefore it would be really handy to have a variable or function to check for this.

**Comment by bencroker** [comment]
Makes sense, will try to fit this in for the next release, thanks for the suggestion!

**Comment by bencroker** [comment]
Looked into this and there are some other things that you should be able to check for to determine whether this is a web request or not, the simplest being whether the `variables` variable is defined.

```twig
{% if variables is defined %}
    This is a web request.
{% endif %}
```

**Comment by bencroker** [comment]
I went ahead and added the `isWebRequest` variable in https://github.com/putyourlightson/craft-campaign/commit/175911ffe176d0bc893c86df2091a6eac2234974 for the next release.

**Comment by bencroker** [comment]
Released in version 1.21.0.
